src/state.py

```diff
@@ -216,13 +216,17 @@ class Artifact(pydantic.BaseModel):
     ssdlc_params: Optional[SSDLCParams] = None
 
     # specific for charms
+    charm: Optional[str] = None
     channel: Optional[str] = None
 
     # specific for OCI images
     image: Optional[str] = None
 
+    # specific for snaps
+    snap: Optional[str] = None
+
     # specific for debs
-    package: Optional[str] = None
+    package: Optional[str] = None  # also for wheels
```

Comment:
I think we can solve this in a simpler way by adding an optional "id" field.
Lacking an id field, we'll use the name to associate artifact spec, downloaded object, and results. If an ID is provided, we'll use that instead.

`name` = the name of the artifact (whatever type it is)
`id` = unique (within a manifest file) identifier for the artifact